### PR TITLE
AJ-1670: MDC requestId for Quartz jobs

### PR DESCRIPTION
**Pull request by davidangb**
Persist the MDC requestId into Quartz job context. Within the Quartz job, retrieve the id from context and put it into the new thread's MDC.

This screenshot of local logs shows the same requestId across threads. In the highlighted rectangle, the first column is the thread name and the second is the request id:
Screenshot 2024-03-08 at 5 04 27 PM

compare to a screenshot from before this PR:
Screenshot 2024-03-03 at 8 32 55 AM

### CODE REVIEW

File: service/src/main/java/org/databiosphere/workspacedataservice/jobexec/QuartzJob.java [hunk 1]
```diff
@@ -50,6 +52,15 @@ public void execute(JobExecutionContext context) throws org.quartz.JobExecutionE
     // retrieve jobId
     UUID jobId = UUID.fromString(context.getJobDetail().getKey().getName());
 
+    // try to retrieve MDC id from job context and add to this thread; don't fail if this errors out
+    try {
+      String requestId =
+          getJobDataString(context.getMergedJobDataMap(), MDCServletRequestListener.MDC_KEY);
+      MDC.put(MDCServletRequestListener.MDC_KEY, requestId);
+    } catch (Exception e) {
```
Comment (jladieu) [inline service/src/main/java/org/databiosphere/workspacedataservice/jobexec/QuartzJob.java:60]: Just curious in what ways it _does_ error out?  Has this bit us before?

**Review by jladieu — APPROVED** [review]
Looks good to me, approved as is...but there is a reasonable looking Sonar warning too.

File: service/src/main/java/org/databiosphere/workspacedataservice/service/MDCServletRequestListener.java [hunk 2]
```diff
@@ -25,7 +25,7 @@
 public class MDCServletRequestListener implements ServletRequestListener {
```
Comment (calypsomatic) [inline service/src/main/java/org/databiosphere/workspacedataservice/service/MDCServletRequestListener.java:26]: Huh didn't even realize/remember we had this class

**Review by calypsomatic — APPROVED** [review]
LGTM

**Review by calypsomatic — APPROVED** [review]

**Review by jladieu — APPROVED** [review]

**Pull request merged**